cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -241,3 +243,61 @@ const char *sc_umount_cmd(char *buf, size_t buf_size, const char *target,
 
 	return buf;
 }
+
+void sc_do_mount(const char *source, const char *target,
+		 const char *fs_type, unsigned long mountflags,
+		 const void *data)
+{
+	char buf[10000];
+	const char *mount_cmd = NULL;
+
+	void ensure_mount_cmd() {
+		if (mount_cmd == NULL) {
+			mount_cmd = sc_mount_cmd(buf, sizeof buf, source,
+						 target, fs_type, mountflags,
+						 data);
+		}
+	}
+
+	if (sc_is_debug_enabled()) {
```

Comment:
I must have misunderstood then. I didn't think we agreed to compile-time only debug. If this is the case let's discuss this further because I would find this much more complicated and less useful for actual analysis.

I am also somewhat sceptical about not using this when we are about to die. This is exactly the sort of message that we will get in bug reports. Having no useful data in there will just make everything harder to analyse. What is the reason you would not like to format that during when we are about to die?